pkgs/development/libraries/mesa/generic.nix

```diff
@@ -212,17 +212,20 @@ self = stdenv.mkDerivation {
     "-Dclang-libdir=${llvmPackages.clang-unwrapped.lib}/lib"
   ] ++ lib.optional enablePatentEncumberedCodecs
     "-Dvideo-codecs=h264dec,h264enc,h265dec,h265enc,vc1dec"
-  ++ lib.optional (vulkanLayers != []) "-D vulkan-layers=${builtins.concatStringsSep "," vulkanLayers}";
+  ++ lib.optional (vulkanLayers != []) "-D vulkan-layers=${builtins.concatStringsSep "," vulkanLayers}"
+  ++ lib.optionals (lib.versionAtLeast version "23.1.0") [
+    # does not make any sense
+    "-Dandroid-libbacktrace=disabled"
```

Comment:
nit: I'm not really a fan of that comment tbh because it makes a statement without any explanation/reasoning.
I assume `android-libbacktrace` is only useful for / only makes sense on Android?
Something like "# Only useful on Android:" would be a bit better IMO (but the current comment should be fine as well as `android` is in the name of the flag).

Makes me wonder why it's enabled by default though (is it even enabled on more platforms than `aarch64`?). Would be interesting if there's an upstream issue regarding this. (The Arch package sets that flag as well FWIW: https://gitlab.archlinux.org/archlinux/packaging/packages/mesa/-/blob/main/PKGBUILD)

Reply:
Upstream has heuristics for enabling things that make sense on the platform where it is being built.
However we (and arch too?) use the meson `auto_features` flag (enabled by default in the meson setup-hook) which tries to automatically enable all possible flags.
I also ran into this when attempting to build on darwin, because there it enables lots of features that don't make sense on darwin.

Reply:
Ah, that makes sense, thanks for the explanation! :)